kola/tests/misc/selinux.go

```diff
@@ -17,52 +17,150 @@ package misc
+		Distros:     []string{"cl", "rhcos"},
+	})
+
+	register.Register(&register.Test{
+		Run:         SelinuxBooleanPersist,
+		ClusterSize: 1,
+		Name:        "rhcos.selinux.boolean.persist",
+		Distros:     []string{"rhcos"},
+	})
+
+	register.Register(&register.Test{
+		Run:         SelinuxManage,
+		ClusterSize: 1,
+		Name:        "rhcos.selinux.manage",
+		Distros:     []string{"rhcos"},
+	})
 }
 
-// SelinuxEnforce checks that some basic things work after `setenforce 1`
-func SelinuxEnforce(c cluster.TestCluster) {
-	m := c.Machines()[0]
+type cmdCheckOutput struct {
+	cmdline     string // command to run
+	checkoutput bool   // should output be checked
+	match       string // regex used to match output from command
+}
 
-	for _, cmd := range []struct {
-		cmdline     string
-		checkoutput bool
-		output      string
-	}{
-		{"getenforce", true, "Permissive"},
-		{"sudo setenforce 1", false, ""},
-		{"getenforce", true, "Enforcing"},
-		{"systemctl --no-pager is-active system.slice", true, "active"},
-		{"sudo cp --remove-destination $(readlink -f /etc/selinux/config) /etc/selinux/config", false, ""},
-		{"sudo sed -i 's/SELINUX=permissive/SELINUX=enforcing/' /etc/selinux/config", false, ""},
-	} {
+// testSelinuxCmds will run a list of commands, optionally check their output, and
+// ultimately reboot the host
+func testSelinuxCmds(c cluster.TestCluster, m platform.Machine, cmds []cmdCheckOutput) {
+	for _, cmd := range cmds {
 		output := c.MustSSH(m, cmd.cmdline)
 
-		if cmd.checkoutput && string(output) != cmd.output {
-			c.Fatalf("command %q has unexpected output: want %q got %q", cmd.cmdline, cmd.output, string(output))
+		if cmd.checkoutput {
+			match := regexp.MustCompile(cmd.match).MatchString(string(output))
+			if !match {
+				c.Fatalf("command %q has unexpected output: tried to match regexp %q, output was %q", cmd.cmdline, cmd.match, string(output))
+			}
 		}
 	}
 
 	err := m.Reboot()
 	if err != nil {
 		c.Fatalf("failed to reboot machine: %v", err)
 	}
+}
+
+// SelinuxEnforce checks that some basic things work after `setenforce 1`
+func SelinuxEnforce(c cluster.TestCluster) {
+	cmdList := []cmdCheckOutput{
+		{"getenforce", true, "Enforcing"},
+		{"sudo setenforce 0", false, ""},
+		{"getenforce", true, "Permissive"},
+		{"sudo setenforce 1", false, ""},
+		{"getenforce", true, "Enforcing"},
+		{"systemctl --no-pager is-active system.slice", true, "active"},
+		{"sudo cp /etc/selinux/config{,.old}", false, ""},
+		{"sudo sed -i 's/SELINUX=permissive/SELINUX=enforcing/' /etc/selinux/config", false, ""},
+	}
+
+	m := c.Machines()[0]
 
-	o := c.MustSSH(m, "which getenforce")
-	c.Logf(string(o))
+	testSelinuxCmds(c, m, cmdList)
 
 	output := c.MustSSH(m, "getenforce")
 
 	if string(output) != "Enforcing" {
-		c.Fatalf("command \"getenforce\" has unexpected output: want \"Enforcing\" got %q", string(output))
+		c.Fatalf("command \"getenforce\" has unexpected output: want \"Enforcing\", got %q", string(output))
```

Comment:
nit: If you want to avoid the escaping you can write this like:

```golang
c.Fatalf(`command "getenforce" has unexpected output: want "Enforcing", got %q`, string(output))
```

Reply:
or just do another `%q` & add `"Enforcing"` to the parameters